src/test/java/org/kainos/ea/service/JobRolesDataTest.java

```diff
@@ -28,33 +31,167 @@ public class JobRolesDataTest {
+                "jobspec.com",
+                "Trainee",
+                6,
+                "Engineering",
+                1,
+                "Engineering");
+
+        Mockito.when( databaseConnector.getConnection() ).thenReturn( conn );
+        Mockito.when( jobRolesData.getJobRole( conn, 1 ) ).thenReturn( expectedResult );
+
+        JobRoleResponse result = jobsService.getJobRole( 1 );
+
+        Assertions.assertEquals(expectedResult, result);
+    }
+
+    @Test
+    void getJobRole_shouldThrowSQLException_whenJobsRolesThrowsSQLException() throws DatabaseConnectionException, SQLException {
+
+        Mockito.when( databaseConnector.getConnection() ).thenReturn( conn );
+        Mockito.when( jobRolesData.getJobRole( conn, 1 ) ).thenThrow( SQLException.class );
+
+        Assertions.assertThrows( SQLException.class,
+                () -> jobsService.getJobRole( 1 ) );
+    }
+
+    @Test
+    void getJobRole_shouldThrowDatabaseConnectionException_whenJobsRolesThrowsDatabaseConnectionException() throws SQLException, DatabaseConnectionException {
+
+        Mockito.when( databaseConnector.getConnection() ).thenReturn( conn );
+        Mockito.when( jobRolesData.getJobRole( conn, 1 ) ).thenThrow( DatabaseConnectionException.class );
+
+        Assertions.assertThrows( DatabaseConnectionException.class,
+                () -> jobsService.getJobRole( 1 ) );
+    }
+
+    @Test
+    void getJobRole_shouldThrowDataNotFoundException_whenJobsRolesThrowsDataNotFoundException() throws DatabaseConnectionException, SQLException, DataNotFoundException {
+
+        Mockito.when( databaseConnector.getConnection() ).thenReturn( conn );
+        Mockito.when( jobRolesData.getJobRole( conn, 1 ) ).thenThrow( DataNotFoundException.class );
+
+        Assertions.assertThrows( DataNotFoundException.class,
+                () -> jobsService.getJobRole( 1 ) );
+    }
+
+    @Test
+    void updateJobRole_shouldReturnTrue_whenJobRoleInsertSuccessful() throws DatabaseConnectionException, SQLException, DataNotFoundException {
+
+        JobRoleRequest jobRoleRequest = new JobRoleRequest(
+                "Software Engineer",
+                "This is a description for Software Engineer",
+                "As a Software Engineer (Associate) in Kainos, you’ll be responsible for developing high quality solutions which delight our customers and impact the lives of users worldwide. You’ll do this whilst learning about new technologies and approaches, with talented colleagues that will help you to learn, develop and grow.",
+                "jobspec.com",
+                6,
+                1,
+                2 );
+
+        Mockito.when( databaseConnector.getConnection() ).thenReturn( conn );
+        Mockito.when( jobRolesData.updateJobRole( conn, 1, jobRoleRequest )).thenReturn( true );
+
+        String result = jobsService.updateJobRole( 1, jobRoleRequest );
+
+        Assertions.assertEquals( "role has been updated", result );
```

Comment:
shouldn't the name of the function be updateJobRole_shouldReturnRoleHasBeenUpdated_whenJobRoleInsertSuccessful() ?